### DISCUSSION TITLE
Enable polygons to show up on the map based on results from PyWPS server

### DISCUSSION THREAD

**Discussion by eravan1** [TerriaJS/terriajs#6693] [Q&A] [answered]
Hi TerriaJS supporters,
Is there a way to make polygons or lines show up on the map based on query results coming back from a PyWPS server (or Geoserver)? 

I added some controls in Data Catalog to allow users to draw a polygon on the map, set a buffer distance and to return one or more points within that area. I was able to send this to a PyWPS server as input. The results came back as a geojson file but the points and buffer would not show up on the map. The returned results did create a workbench item on the side panel.

Thanks for any help on this.

**Comment by zoran995** [comment, marked as answer]
Hi! This was asked a while ago, but it might still be useful for others. To display vector data generated by a WPS server, you can have the WPS service return a response that adds a new item to the workbench. This will automatically show the resulting data on the map.

You can find more details in this discussion, along with a sample pywps service:
https://github.com/TerriaJS/terriajs/discussions/7610#discussioncomment-12971301
